sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -295,6 +295,14 @@ int getAttribute(pi_device device, CUdevice_attribute attribute) {
 }
 /// \endcond
 
+template <typename T,
+          typename = std::enable_if_t<std::numeric_limits<
+              std::remove_reference_t<std::remove_cv_t<T>>>::is_integer>>
+constexpr inline bool isPowerOf2(T value) noexcept {
```

Comment:
It's only ever used in one place, I'd be tempted to fold it into `guessLocalWorkSize`, maybe as a lambda in the body?